lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2693,83 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onViewFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent viewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+///
+/// This value will be tipically passed to the [PlatformDispatcher.onViewFocusChange]
+/// callback.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+    required this.focusDirection,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState focusState;
+
+  /// The current [ViewFocusDirection] of the [FlutterView] with the given [viewId].
```

Comment:
Same here, instead of "current", consider something like "The direction in which the focus changed".